packages/orchestration/src/orchestration-stream.ts

```diff
@@ -51,53 +49,87 @@ export class OrchestrationStream<Item> extends SseStream<Item> {
   /**
    * @internal
    */
-  static async *_processFinishReason(
+  static async *_processToolCalls(
     stream: OrchestrationStream<OrchestrationStreamChunkResponse>,
     response?: OrchestrationStreamResponse<OrchestrationStreamChunkResponse>
   ): AsyncGenerator<OrchestrationStreamChunkResponse> {
     for await (const chunk of stream) {
-      chunk.data.orchestration_result?.choices.forEach(
-        (choice: LlmChoiceStreaming) => {
-          const choiceIndex = choice.index;
-          if (choiceIndex >= 0) {
-            const finishReason = chunk.getFinishReason(choiceIndex);
-            if (finishReason) {
-              if (response) {
-                response._getFinishReasons().set(choiceIndex, finishReason);
+      chunk.data.orchestration_result?.choices.forEach(choice => {
+        const choiceIndex = choice.index;
+        if (choiceIndex >= 0) {
+          const toolCallsChunks = chunk.getToolCalls(choiceIndex);
+          if (toolCallsChunks) {
+            if (response) {
```

Comment:
[pp] Is checking `choiceIndex >= 0` necessary? If it is -1 we don't care right? Because `chunk.getToolCalls(choiceIndex)` will return `undefined` anyway. So I feel we can merge this three `if`s into one.

Reply:
Or even better, we can use `?` for the next `toolCallAccumulators` and merge `if (!toolCallAccumulators)` as well.